tools/harvest_source_import/tests/test_list_sources.py

```diff
@@ -0,0 +1,92 @@
+""" test the list of harvest sources
+ - Add this to VCR cassette: python -m pytest -s --vcr-record=all tests/test_list_sources.py
+"""
+
+import pytest
+import vcr
+from time import sleep
+from remote_ckan.lib import RemoteCKAN
+import urllib.parse as urlparse
+
+@pytest.mark.vcr()
+def test_list_ckan_sources():
+    """ Test the list of sources """
+
+    ckan = RemoteCKAN(url='https://catalog.data.gov')
+    total = 0
+    expected_names = ['doi-open-data', 'test-2016']
+
+    results = {}
+    for hs in ckan.list_harvest_sources(source_type='ckan'):
+        total += 1
+        assert hs['source_type'] == 'ckan'
+        assert hs['name'] in expected_names
+        results[hs['name']] = hs
+
+    assert total == 2
+    assert results['doi-open-data']['url'] == 'https://data.doi.gov'
+    assert results['doi-open-data']['status']['job_count'] == 1
+
+
+def test_requests_sent_for_ckan_sources():
+    cass = vcr.cassette.Cassette.load(path='tests/cassettes/test_list_ckan_sources.yaml')
+    search_request = cass.requests[0]
```

Comment:
How do you know that the request's we're inspecting is the actual request and not the request listed in the [test data](https://github.com/GSA/catalog.data.gov/pull/127/files#diff-aafb6264c1e5675c7ea4ccaab2c0bf43R123)?

I would expect to see a `RemoteCKAN.list_harvest_sources` call within the test. Otherwise it looks like we're not running any code under test.

Reply:
I assume the list of requests in the `test_list_ckan_sources.yaml` file are ordered. The first one is the `package_search` for the first page of results.

The `list_harvest_sources` function [is called](https://github.com/GSA/catalog.data.gov/blob/bugfix/harvest-source-type/tools/harvest_source_import/tests/test_list_sources.py#L20) in the tests. This call saved the yaml file (with the same name as the test function).

You can delete the cassettes and run the test again and it will continue working.

Reply:
I will merge, we can add new changes in a new PR if needed.